src/client/Client.cc

```diff
@@ -8042,6 +8042,8 @@ int Client::_do_setattr(Inode *in, struct ceph_statx *stx, int mask,
   int res;
   {
     std::string path;
+    if (metadata.count("root"))
+      path.append(metadata["root"].data(), metadata["root"].length());
     res = in->make_path_string(path);
```

Comment:
I don't think this is a safe interface, to require the user to just know to pre-fill the path string. The description for ```make_path_string`` says "make a filepath suitable for mds auth access check", and if you need to manipulate it separately, that's not great.

The Inode already has a Client*, so we should be able to grab anything we need from there, right? (Also, is using metadata["root"] really the best source for that? It probably works, but I thought we had it somewhere more authoritative.)

Reply:
Okay, I will add one helper in `Client` and to simplify this and at the same time to export the symbol via libcephfs.